lock/lock.go

```diff
@@ -307,3 +307,37 @@ func randomizedCopy(keys []string) []string {
 	})
 	return randomizedKeys
 }
+
+func (l locker) GetCurrentLock(ctx context.Context, key string) (*Lock, error) {
+	getInput := dynamodb.GetItemInput{
+		Key: map[string]*dynamodb.AttributeValue{
+			partitionKeyName: &dynamodb.AttributeValue{
+				S: &key,
+			},
+		},
+		TableName:      aws.String(l.tableName),
+		ConsistentRead: aws.Bool(true),
+	}
+
+	result, err := l.dynamoDBAPI.GetItemWithContext(ctx, &getInput)
+
+	if err != nil {
+		l.logger.ErrorD("unexpected-dynamo-error", logger.M{
```

Comment:
nit: do we still want to log here since we're returning the error? If we do then maybe we also want to log for the unmarshal error below

Reply:
I _think_ I was following the pattern that was established for other dynamo access - I'm not attached to this though.

https://github.com/Clever/dynamodb-lock-go/blob/b4feaddeb8ae4d0de9e7d57972b29f899dad6130/lock/lock.go#LL217-L225

and 

https://github.com/Clever/dynamodb-lock-go/blob/b4feaddeb8ae4d0de9e7d57972b29f899dad6130/lock/lock.go#L186-L189

Reply:
I think when I made this library, my intention was to log somewhat verbosely and the user can control things by the `logger` argument to https://github.com/Clever/dynamodb-lock-go/blob/master/lock/interface.go#L75. The way things are done in this PR is consistent with that pattern, but I'm not super attached to the pattern.